tests/Microsoft.Data.Analysis.Tests/DataFrame.IOTests.cs

```diff
@@ -104,5 +104,140 @@ Stream GetStream(string streamData)
             Assert.Equal("", nullRow[5]);
             Assert.Null(nullRow[6]);
         }
+
+        [Fact]
+        public void TestReadCsvWithPipeSeparator()
+        {
+            string data = @"vendor_id|rate_code|passenger_count|trip_time_in_secs|trip_distance|payment_type|fare_amount
+CMT|1|1|1271|3.8|CRD|17.5
```

Comment:
So, I looked at the input in the existing tests, is there a reason you didn't add the `||||||` line here?

Reply:
Similar question with the `;` separator

Reply:
I used the same input from the first test. Is that row needed to just make sure it doesn't cause an exception?

Reply:
That's one the reasons yes. The other being that for a `StringDataFrameColumn`, empty values are valid. For other column types, empty values are considered as `null` values and increase the `NullCount`.

Reply:
Great points! Updated to add the empty values column.